source/schedulexmlnode.d

```diff
@@ -0,0 +1,249 @@
+module scheduleXmlNode;
+
+import std.datetime;
+import kxml.xml : XmlNode;
+import std.array : empty, front;
+
+import xmlconstants;
+import xmlparserexceptions;
+
+public: 
+struct ScheduleXmlNode {
```

Comment:
I am in doubt if this whole structure is a good idea. To me it looks like we are tying ourselves even closer to the XML library. The fact that we are about to switch the XML library does not seem to work well with this.

What is the rationale for adding the struct and simply keep the functions as free functions?